core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -225,6 +225,16 @@ public static LoadTableResponse loadTable(Catalog catalog, TableIdentifier ident
     Table table = catalog.loadTable(ident);
 
     if (table instanceof BaseTable) {
+      // This gives
+      // TestRESTCatalog > testCompleteReplaceTransaction() FAILED
+      // java.lang.AssertionError: Table should have correct number of previous metadata locations expected:<1> but was
+      //   :<2>
+      // TableMetadata metadataWithoutLocation = ((BaseTable) table).operations().current();
+      // String metadataLocation = ((BaseTable) table).operations().metadataFileLocation("metadataV1.json");
+      // TableMetadata metadata = TableMetadata.buildFrom(metadataWithoutLocation)
+      //     .withMetadataLocation(metadataLocation)
+      //     .build();
+      // String metadataLocation = ((BaseTable) table).operations().current().previousFiles();
       return LoadTableResponse.builder()
           .withTableMetadata(((BaseTable) table).operations().current())
```

Comment:
This `TableMetadata` has no metadataLocation value on it. Tracking down where this is built and will update it.

Reply:
It actually does. There's some internal usage of `LoadTableResponse` where the `metadataLocation` isn't known (e.g. during `stageCreate`).

Removing the requirement that the result of the builder have `metadataLocation` at all times fixes this.